src/python/setup.py.in

```diff
@@ -24,18 +25,27 @@ package_name = '@TARGET_NAME@'
 
 def _onnxruntime_dependency() -> str:
     dependency = None
+    # Use dev version as default since CI tests use nightly version for testing
+    ort_version = os.environ.get("ONNXRUNTIME_VERSION", "1.19.0.dev20240805002")
+    is_nightly = True if "dev" in ort_version else False
+
     if package_name == "onnxruntime-genai":
-        dependency = "onnxruntime"
+        dependency = "onnxruntime" if not is_nightly else "ort-nightly"
+
+        import platform
+        # win arm64 whls are only available in onnxruntime-qnn
+        if platform.machine() == "ARM64" and sys.platform.startswith("win"):
+            dependency = "onnxruntime-qnn" if not is_nightly else "ort-nightly-qnn"
     elif package_name == "onnxruntime-genai-cuda":
-        dependency = "onnxruntime-gpu"
+        dependency = "onnxruntime-gpu" if not is_nightly else "ort-nightly-gpu"
     elif package_name == "onnxruntime-genai-directml":
-        dependency = "onnxruntime-directml"
+        dependency = "onnxruntime-directml" if not is_nightly else "ort_nightly-directml"
```

Comment:
Should it be ort-nightly-directml? Not sure about the name.

Reply:
yes, you're right. Thanks for catching this.
In python _ and - should be treated the same. But let me correct this.